include/rocksdb/utilities/transaction_db.h

```diff
@@ -202,6 +202,13 @@ struct KeyLockInfo {
   bool exclusive;
 };
 
+struct RangeLockInfo {
+  Endpoint start;
+  Endpoint end;
+  std::vector<TransactionID> ids;
+  bool exclusive;
```

Comment:
Does range locking support shared locks? Or is this here just for future extensibility?

Reply:
For future extensibility.